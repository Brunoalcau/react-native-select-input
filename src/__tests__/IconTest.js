
import React from 'react';
import { shallow } from 'enzyme';
import Icon, { ARROW_ICON_TYPES } from '../Icon';

describe('Icon', () => {
  beforeEach(() => {
    this.props = {
      type: ARROW_ICON_TYPES.UP,
      loading: false,
      disabled: false,
    };
  });

  it('renders arrow up correctly', () => {
    const wrapper = shallow(<Icon {...this.props} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders arrow down correctly', () => {
    const wrapper = shallow(
      <Icon
        {...this.props}
        type={ARROW_ICON_TYPES.DOWN}
      />,
    );

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
