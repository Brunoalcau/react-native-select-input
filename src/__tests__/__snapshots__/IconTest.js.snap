// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders arrow down correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/icons/arrowDown.png",
    }
  }
  style={
    Array [
      Object {
        "height": 5,
        "marginTop": 9,
        "width": 8,
      },
      null,
    ]
  }
/>
`;

exports[`Icon renders arrow up correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/icons/arrowUp.png",
    }
  }
  style={
    Array [
      Object {
        "height": 5,
        "marginTop": 9,
        "width": 8,
      },
      null,
    ]
  }
/>
`;
